### PR TITLE
[Concept Entry] PyTorch nn

### LINKED ISSUE #4826: [Concept Entry] PyTorch nn
### Reason/inspiration (optional)

We would like a new entry on the `nn` concept under PyTorch. The entry should go in a new file under `docs/content/pytorch/concepts/nn/nn.md`.

The entry should include:

- An introduction to the concept
- A ‘Syntax’ section that provides the syntax for the concept (if applicable)
- An ‘Example’ section that provides an example demonstrating the concept in use
- A ‘Codebyte Example’ section that provides a codebyte example demonstrating the concept in use

Please refer to the [concept entry template](https://github.com/Codecademy/docs/blob/main/documentation/concept-entry-template.md), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Concept Entrty for Existing Topic

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [ ] Issue is assigned

### PR DESCRIPTION
**Pull request by andersooi**
### Description

- Added new concept entry for PyTorch nn

### Issue Solved

Closes #4826  

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/pytorch/concepts/nn/nn.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+Tags:
+  - 'AI'
+  - 'Neural Networks'
+CatalogContent:
+  - 'paths/data-science'
+---
+
+## Introduction
+
+The **`nn module`** in PyTorch provides the foundation to build and train neural network models. It contains various layers, activation functions, loss functions and optimisation algorithms that enable the creation of complex deep learning models. The **`nn module`** is highly customisable, enabling developers to construct their own neural network architectures.
```
Comment (ishg-153) [inline content/pytorch/concepts/nn/nn.md:17]:
```suggestion
The **`nn module`** in PyTorch provides the foundation to build and train neural network models. It contains various layers, activation functions, loss functions, and optimization algorithms that enable the creation of complex deep-learning models. The `nn module` is highly customizable, enabling developers to construct neural network architectures.
```
Do not use personal pronouns

File: content/pytorch/concepts/nn/nn.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  def forward(self, x):
+    x = torch.relu(self.fc1(x))
+    x = self.fc2(x)
+    return x
+
+# Create a model instance
+model = SimpleNN()
+```
+
+## Codebyte Example
```
Comment (ishg-153) [inline content/pytorch/concepts/nn/nn.md:64]: You can remove this section as codecademy workspace is not compatible with pytorch yet, so this codebyte won't run.

**Review by ishg-153 — CHANGES_REQUESTED** [review]
@andersooi Please make these changes

File: content/pytorch/concepts/nn/nn.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+Tags:
+  - 'AI'
+  - 'Neural Networks'
+CatalogContent:
+  - 'paths/data-science'
+---
+
+## Introduction
+
+The **`nn module`** in PyTorch provides the foundation to build and train neural network models. It contains various layers, activation functions, loss functions, and optimization algorithms that enable the creation of complex deep-learning models. The `nn module` is highly customizable, enabling developers to construct neural network architectures.
```
Comment (avdhoottt) [inline content/pytorch/concepts/nn/nn.md:17]:
```suggestion
The `**torch.nn**` module in PyTorch provides the foundation for building and training neural network models. It includes a wide range of pre-built layers, activation functions, loss functions, and other components necessary for creating complex deep-learning models. The `torch.nn` module is highly flexible and customizable, allowing developers to design and implement neural network architectures tailored to specific tasks.
```

File: content/pytorch/concepts/nn/nn.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+class NeuralNetwork(nn.Module):
+  def __init__(self):
+    super(NeuralNetwork, self).__init__()
+    # Define layers, activations, and other components
+
+  def forward(self, x):
+    # Implement forward pass of model
+    return output
+
+  # Define other methods where necessary
```
Comment (avdhoottt) [inline content/pytorch/concepts/nn/nn.md:36]:
```suggestion
import torch.nn as nn

# Define a neural network model
class NeuralNetwork(nn.Module):
    def __init__(self):
        super(NeuralNetwork, self).__init__()
        # Define layers, activations, and other components here
        # For example: self.layer1 = nn.Linear(in_features, out_features)

    def forward(self, x):
        # Implement the forward pass of the model
        # For example: x = self.layer1(x)
        return x  # Replace 'x' with the final output after applying all layers

    # Define other methods if necessary
```

File: content/pytorch/concepts/nn/nn.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        return x  # Replace 'x' with the final output after applying all layers
+
+    # Define other methods if necessary
+```
+
+- `NetworkNeural`: A custom neural network class that subclasses the `nn.Module`.
+
+## Example
+
+The following example demonstrates how to define a simple neural network model using the `nn.Module`:
```
Comment (avdhoottt) [inline content/pytorch/concepts/nn/nn.md:45]:
```suggestion
The following example demonstrates how to define a simple neural network model using the `nn.Module` with two fully connected layers, apply a ReLU activation function to the first layer's output, and create an instance of the model:
```

**Review by avdhoottt — APPROVED** [review]
LGTM!!

File: content/pytorch/concepts/nn/nn.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+Description: 'Build and train neural network models with a range of layers, activations, loss functions and optimisation algorithms.'
+Subjects:
+  - 'Machine Learning'
+  - 'Computer Science'
+  - 'AI'
+Tags:
+  - 'AI'
+  - 'Neural Networks'
+CatalogContent:
+  - 'paths/data-science'
```
Comment (mamtawardhani) [inline content/pytorch/concepts/nn/nn.md]:
```suggestion
  - 'intro-to-py-torch-and-neural-networks'
  - 'py-torch-for-classification'
```

**Review by mamtawardhani — APPROVED** [review]
Hey @andersooi thank you for contributing to Codecademy Docs! 😄 

The entry is ready to be merged! 🚀
